pkg/artifactory/provider.go

```diff
@@ -65,6 +65,7 @@ func Provider() *schema.Provider {
 			"artifactory_keypair":                    resourceArtifactoryKeyPair(),
 			"artifactory_local_repository":           resourceArtifactoryLocalRepository(),
 			"artifactory_local_nuget_repository":     resourceArtifactoryLocalNugetRepository(),
+			"artifactory_local_npm_repository":       resourceArtifactoryLocalNpmRepository(),
```

Comment:
Let's have a call. I thought I had explained this yesterday. Given how many repos are literally identical to generic repositories, that what we should do is:

```go
"artifactory_local_npm_repository" : resourceArtifactoryLocalGenericRepository("npm")
```

Then, adding support for all the different repo types that are masquerading as "generic" is trivial

Reply:
Made changes to cover all repos that are similar to generic. And, updated this PR.
Let me know if i can push this under a new PR instead.